packages/@ember/-internals/glimmer/lib/utils/managers.ts

```diff
@@ -57,20 +66,26 @@ function getManager<D extends ManagerDelegate>(
 }
 
 function getManagerInstanceForOwner<D extends ManagerDelegate>(
-  owner: Owner,
-  factory: ManagerFactory<D>
+  owner: Owner | undefined,
+  factory: ManagerFactory<Owner, D>
 ): D {
-  let managers = MANAGER_INSTANCES.get(owner);
+  let managers;
 
-  if (managers === undefined) {
-    managers = new WeakMap();
-    MANAGER_INSTANCES.set(owner, managers);
+  if (owner === undefined) {
+    managers = UNDEFINED_MANAGER_INSTANCES;
+  } else {
+    managers = OWNER_MANAGER_INSTANCES.get(owner);
+
+    if (managers === undefined) {
+      managers = new WeakMap();
+      OWNER_MANAGER_INSTANCES.set(owner, managers);
+    }
   }
 
   let instance = managers.get(factory);
 
   if (instance === undefined) {
-    instance = factory(owner);
+    instance = factory(owner!);
```

Comment:
How do you know that owner exists here?

Reply:
It may actually be `undefined`, in the case of helper managers, but for modifier/component managers it will always exist at this point (since you are not currently allowed to pass an `undefined` owner to `getModifierManager` or `getComponentManager`). The types here are a bit messy, but I think it _should_ be fine.